core/src/spotify_id.rs

```diff
@@ -136,15 +137,15 @@ impl SpotifyId {
 
     /// Returns the `SpotifyId` as a base16 (hex) encoded, `SpotifyId::SIZE_BASE16` (32)
     /// character long `String`.
-    pub fn to_base16(&self) -> String {
+    pub fn to_base16(&self) -> Result<String, FromUtf8Error> {
```

Comment:
I prefer this to return the ubiquitous `librespot_core::Error` type. (Wait, is this PR against `dev` or `new-api`? Let's make it the latter.)